src/main/java/com/parasoft/demoapp/service/ItemInventoryService.java

```diff
@@ -96,11 +100,17 @@ private void increaseItemStock(InventoryInfoDTO requestedItem) {
     private String validateItemStockBeforeDecrease(List<InventoryInfoDTO> requestedItems) {
         for (InventoryInfoDTO requestedItem : requestedItems) {
             Long itemId = requestedItem.getItemId();
+            ItemEntity item;
+            try {
+                item = itemService.getItemById(itemId);
+            } catch (Exception e) {
+                item = null;
+            }
             ItemInventoryEntity inventoryItem = itemInventoryRepository.findByItemId(itemId);
-            if (inventoryItem == null) {
-                return String.format("Inventory item with id %d doesn't exist.", itemId);
+            if (item == null || inventoryItem == null) {
+                return "Your order contains invalid items.";
             } else if (inventoryItem.getInStock() < requestedItem.getQuantity()) {
-                return String.format("Inventory item with id %d is out of stock.", itemId);
+                return String.format("%s is out of stock.", item.getName());
             }
```

Comment:
@yshi-parasoft  For changes in this file, maybe is not necessery. But i think the original message expose item id is not good idea. Let me know if we don't need changes in this file.